go-client/examples/basic_query/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/deephaven/deephaven-core/go-client/client"
+	"github.com/deephaven/deephaven-core/go-client/examples/common"
+)
+
+// This example shows how to use the powerful query system,
+// which allows you to perform an arbitrary number of table operations in a single request.
+// It can even create new tables as part of a query,
+// and the interface is more convenient than doing each of the table operations serially.
+func main() {
+	// A context is used to set timeouts and deadlines for requests or cancel requests.
+	// If you don't have any specific requirements, context.Background() is a good default.
+	ctx := context.Background()
+
+	// When starting a client connection, note that the client script language "python"
+	// must match the language the server was started with,
+	// even if we aren't using any scripts.
```

Comment:
```suggestion
	// When starting a client connection, the client script language
	// must match the language the server was started with,
	// even if the client does not execute any scripts.
```
Does `NewClient` return an error if the script lang is wrong?

Reply:
Yes. Currently it just returns a raw gRPC error:
`rpc error: code = FailedPrecondition desc = session type 'groovy' is not supported`
This would be easy to wrap, too, but I'm not sure I want to rely on parsing the error description.